CONTRIBUTING.md

```diff
@@ -0,0 +1,57 @@
+# Building and contributing to TensorFlow Java
+
+## Building
+
+To build all the artifacts, simply invoke the command `mvn install` at the root of this repository (or
+the Maven command of your choice). It is also possible to build artifacts with support for MKL enabled with
+`mvn install -Djavacpp.platform.extension=-mkl` or CUDA with `mvn install -Djavacpp.platform.extension=-gpu`
+or both with `mvn install -Djavacpp.platform.extension=-mkl-gpu`.
+
+When building this project for the first time in a given workspace, the script will attempt to download
+the [TensorFlow runtime library sources](https://github.com/tensorflow/tensorflow) and build of all the native code
+for your platform. This requires a valid environment for building TensorFlow, including the [bazel](https://bazel.build/)
+build tool and a few Python dependencies (please read [TensorFlow documentation](https://www.tensorflow.org/install/source)
+for more details).
+
+This step can take multiple hours on a regular laptop. It is possible though to skip completely the native build if you are
+working on a version that already has pre-compiled native artifacts for your platform [available on Sonatype OSS Nexus repository](#Snapshots).
+You just need to activate the `dev` profile in your Maven command to use those artifacts instead of building them from scratch
+(e.g. `mvn install -Pdev`).
+
+Note that modifying any source files under `tensorflow-core` may impact the low-level TensorFlow bindings, in which case a
```

Comment:
Actually, I think a full native build is only required if we modify the op generator sources. It should be safe to reuse the pre-compiled native artifacts when modifying any of the Java source files

Reply:
Agreed, that was in the readme so I didn't want to change it.  I'll update it.